Orange/widgets/visualize/tests/test_owsieve.py

```diff
@@ -22,10 +24,52 @@ def setUpClass(cls):
 
         cls.signal_name = "Data"
         cls.signal_data = cls.data
+        cls.titanic = Table("titanic")
+        cls.iris = Table("iris")
 
     def setUp(self):
         self.widget = self.create_widget(OWSieveDiagram)
 
+    def test_context_settings(self):
+        # Set titanic and check first two attributes on display
+        self.send_signal(self.widget.Inputs.data, self.titanic)
+        self.assertEqual(self.widget.attr_x, self.titanic.domain.class_var)
+        self.assertEqual(self.widget.attr_y, self.titanic.domain.attributes[0])
+        # Change attributes to two different ones
+        self.widget.attr_x = self.titanic.domain.attributes[1]
+        self.widget.attr_y = self.titanic.domain.attributes[2]
+        # Remove data signal
+        self.send_signal(self.widget.Inputs.data, None)
+        self.assertEqual(self.widget.attr_x, None)
+        self.assertEqual(self.widget.attr_y, None)
```

Comment:
Since you are testing `discrete_data` below ... maybe you can add
```
        self.assertIsNone(self.widget.discrete_data)
```

And then you can fix the widget to pass this test. :)